src/main/target/VGOODF722DUAL/target.h

```diff
@@ -0,0 +1,202 @@
+#define USE_MAG_QMC5883
+#define USE_MAG_LIS3MDL
+*/
+
+//-------------------------------------USART-----------------------------------
+#define USE_VCP
+#define USB_DETECT_PIN          PC14
+#define USE_USB_DETECT
+
+#define USE_UART1
+#define UART1_TX_PIN            PA9
+#define UART1_RX_PIN            PA10
+
+#define USE_UART2
+#define UART2_TX_PIN            PA2
+#define UART2_RX_PIN            PA3
+
+#define USE_UART3
+#define UART3_TX_PIN            PB10
+#define UART3_RX_PIN            PB11
+
+#define USE_UART4
+#define UART4_TX_PIN            PA0
+#define UART4_RX_PIN            PA1
+
+#define USE_UART6
+#define UART6_TX_PIN            PC6
+#define UART6_RX_PIN            PC7
+     
+#define USE_SOFTSERIAL1
+
+#define SERIAL_PORT_COUNT       7
+
+//-------------------------------------ADC METERE------------------------------
+#define USE_ADC
+//#define ADC_INSTANCE                        ADC3
+#define ADC1_DMA_STREAM         DMA2_Stream0
+#define VBAT_ADC_PIN            PC0
+#define CURRENT_METER_ADC_PIN   PC1
+#define RSSI_ADC_PIN            PC2
+#define EXTERNAL1_ADC_PIN       PA4
+//---------------------------------VTX&&CAM------------------------------------
+#define USE_PINIO
+#define PINIO1_PIN              PC8 // VTX power switcher
+#define PINIO2_PIN              PC9 // 2xCamera switcher
+#define USE_PINIOBOX
+//-------------------------------------GPIO-----------------------------------
+#define USE_SPI
+#define USE_SPI_DEVICE_3		//GYRO
+#define SPI3_SCK_PIN            PC10
+#define SPI3_MISO_PIN           PC11
+#define SPI3_MOSI_PIN           PC12
+
+#define USE_SPI_DEVICE_2 // FLASH
+#define SPI2_SCK_PIN            PB13
+#define SPI2_MISO_PIN           PB14
+#define SPI2_MOSI_PIN           PB15
+
+#define USE_SPI_DEVICE_1	//OSD
+#define SPI1_SCK_PIN            PA5
+#define SPI1_MISO_PIN           PA6
+#define SPI1_MOSI_PIN           PA7
+
+#define USE_I2C				//BMP280
+#define USE_I2C_DEVICE_1
+#define I2C_DEVICE              (I2CDEV_1)
+//#define I2C_DEVICE_1            (I2CDEV_1)
+#define I2C1_SCL                PB8
+#define I2C1_SDA                PB9
+
+//-------------------------------------FEATURES-------------------------------
+#define DEFAULT_FEATURES        (FEATURE_OSD | FEATURE_TELEMETRY )
+
+#define DEFAULT_RX_FEATURE     	FEATURE_RX_SERIAL
+#define SERIALRX_PROVIDER       SERIALRX_SBUS
+#define SERIALRX_UART           SERIAL_PORT_USART2
+
+#define DEFAULT_VOLTAGE_METER_SOURCE VOLTAGE_METER_ADC
+#define DEFAULT_CURRENT_METER_SOURCE CURRENT_METER_ADC
+#define CURRENT_METER_SCALE_DEFAULT 179
+//#define VBAT_SCALE_DEFAULT          160
+
+#define USE_ESCSERIAL
+#define USE_SERIAL_4WAY_BLHELI_INTERFACE
+
+//--------------------------------BOARD RESOURCES-----------------------------
+#define TARGET_IO_PORTA 0xffff
+#define TARGET_IO_PORTB 0xffff
+#define TARGET_IO_PORTC 0xffff
+#define TARGET_IO_PORTD 0xffff
+
+#define USABLE_TIMER_CHANNEL_COUNT 9
+#define USED_TIMERS    (TIM_N(1)|TIM_N(3)|TIM_N(4)|TIM_N(9)
```

Comment:
This line has got unbalanced braces - currently the firmware does not even build. Please at least test to make sure the firmware is building **before** opening or updating pull requests.